src/SOS/inc/specialthreadinfo.h

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// ******************************************************************************
+// WARNING!!!: This code is also used by createdump in the runtime repo.
+// See: https://github.com/dotnet/runtime/blob/main/src/coreclr/debug/createdump/specialthreadinfo.h
+// ******************************************************************************
+
+// This defines a workaround to the MacOS dump format not having the OS process
+// and thread ids that SOS needs to map thread "indexes" to thread "ids". The MacOS
+// createdump adds this special memory region at this specific address that is not
+// in the user or kernel address spaces. lldb is find with it.
+
+#define SPECIAL_THREADINFO_SIGNATURE "THREADINFO"
+
+const uint64_t SpecialThreadInfoAddress = 0x7fffffff00000000;
+
+struct SpecialThreadInfoHeader
+{
+    char signature[16];
+    uint32_t pid;
+    uint32_t numThreads;
```

Comment:
I am surprised there is not a `SpecialThreadInfoEntry entries[0]; ` line here.  At least a comment that `SpecialThreadInfoEntry entries[numThreads]` is effectively implied there

Reply:
I guess I could have done that but this is just a copy of the h file in the runtime so I don't really want to change it now.

Reply:
I'll add a comment.